### PR TITLE
Container config recon

### PR DESCRIPTION
**Pull request by coffeepac**
these changes cover the transition from using a container config that hangs off the cluster level to one that hangs off of the nodePool level.  This configuration move challenges how we approach container level configurations in a rather fundamental way.  This should be obvious because I think this PR touches everything in our codebase.  

This approach challenge means that we need to start off by iterating over a list of clusters and passing a list of nodePools down.  it also means that we can no longer attempt to access a global etcd or master configuration because in a single config file there may be several sets of both that have no relation to each other.

Anyway, here's wonderwall.

### CODE REVIEW

**Comment by coffeepac** [comment]
This resolves #272

File: ansible/roles/kraken.cluster_common/tasks/docker-install.yaml [hunk 1]
```diff
@@ -1,60 +1,22 @@
 ---
 - name: Determine which nodes will use which container runtime
   set_fact:
-    custom_etcd: "{{ kraken_config.etcd
-                   | selectattr('nodepool.containerConfig.runtime', 'equalto', 'docker')
-                   | selectattr('nodepool.containerConfig.type', 'equalto', 'tgz')
+    custom_nodes: "{{ cluster.nodePools
```
Comment (joejulian) [inline ansible/roles/kraken.cluster_common/tasks/docker-install.yaml]: cluster.nodePools is a list. docker-install.yaml will need to be looped by nodePools as well as by clusters.

Comment (coffeepac) [inline ansible/roles/kraken.cluster_common/tasks/docker-install.yaml]: docker-install.yaml is being called once per cluster.

I don't think we need to do anything more as selectattr will operate on a list input.  This code isn't changing actually, in the old version both kraken_config.etcd and kraken_config.node are lists

**Comment by mikeln** [comment]
LGTM

**Pull request merged**